### ISSUE TITLE
contract.methods['functionName(address)'](address) is running method with string signature

### ISSUE THREAD

**Issue by TechnicallyWeb3** [ethereum/web3.js#6837]
#### Expected behavior
Expect `contract.methods['functionName(address)'](address).call();` to call the method with the address signature.

#### Actual behavior
Call happens on the method with string signature instead.

#### Steps to reproduce the behavior
```
// in js file
async function getUserData(address) {
    return await userContract.methods['getUserData(address)'](address).call();
}

// in solidity file
function getUserData(address userAddress) public view returns (UserData memory) {
    return _userData[_idOfAddress[userAddress]];
}

function getUserData(string memory userName) public view returns (UserData memory) {
    return  _userData[_idOfName[userName]];
}
```

1. load contract (live on sepolia testnet @ 0x6f380E5d68D7211f305976f86c66a22Ce17C3054)
2. in a browser with the js loaded run "await getUserData('0xF174CC549e39545a44041bbf3Daf94E5b6DD7F54')" in console
3. in a browser with the js loaded run "await getUserData('technicallywebdev')" in console
4. expect first to log userData object, expect 2nd to fail with incorrect type.

#### Logs

output of step 2: 
`{0: 0n, 1: '0x0000000000000000000000000000000000000000', 2: '', 3: '', 4: '', 5: Array(0), __length__: 6, registrationDate: 0n, defaultAddress: '0x0000000000000000000000000000000000000000', userName: '', userBio: '', …}`
output of step 3: 
`{0: 1709018832n, 1: '0xF174CC549e39545a44041bbf3Daf94E5b6DD7F54', 2: 'technicallydev', 3: 'Hello World!', 4: '/assets/images/logo.png', 5: Array(1), __length__: 6, registrationDate: 1709018832n, defaultAddress: '0xF174CC549e39545a44041bbf3Daf94E5b6DD7F54', userName: 'technicallydev', userBio: 'Hello World!', …}`

#### Environment

run client-side via html import. 
npm: 9.8.0, node: 18.6.1 web3 version: https://cdn.jsdelivr.net/npm/web3@latest/dist/web3.min.js
the full project is available on https://github.com/TechnicallyWeb3/technicallyweb3.com/commit/2cddcb69b351271dd680345377000f941329030f

**Comment by mconnelly8** [comment]
Thanks for submitting all the details @TechnicallyWeb3.  The team will review this soon.

**Comment by TechnicallyWeb3** [comment]
### Additional details:

#### Expected Behavior:

Expect the encodeABI() function to encode the correct signature.

#### Actual Behavior:

encodeABI() returns the wrong signature for selected functions. 

#### Steps to reproduce:
```
// js file
async function getUserData(address) {
    console.dir(userContract.methods['getUserData(address)'](connectedAddress).encodeABI());
    console.dir(userContract.methods['0xffc9896b'](connectedAddress).encodeABI());
    console.dir(userContract.methods.getUserData('0xedE6f94B27D4E87a5C91b43D3F6CA428636E112D').encodeABI());
    console.dir(userContract.methods['isUser(address)'](connectedAddress).encodeABI());
    console.dir(userContract.methods['0x4209fff1'](connectedAddress).encodeABI());
    console.dir(userContract.methods.isUser('0xedE6f94B27D4E87a5C91b43D3F6CA428636E112D').encodeABI());
    // process.stdout.write(JSON.stringify(userContract.methods['getUserData(address)']) + '\n');
    return await userContract.methods['getUserData(address)'](address).call();
}
```

// output
14:06:03.628 scripts.js:71 **0x1fef580c**000000000000000000000000ede6f94b27d4e87a5c91b43d3f6ca428636e112d
14:06:03.628 scripts.js:72 **0x1fef580c**000000000000000000000000ede6f94b27d4e87a5c91b43d3f6ca428636e112d
14:06:03.628 scripts.js:73 **0x1fef580c**000000000000000000000000ede6f94b27d4e87a5c91b43d3f6ca428636e112d
14:06:03.629 scripts.js:74 **0x05bbd7e**00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002a30786564453666393442323744344538376135433931623433443346364341343238363336453131324400000000000000000000000000000000000000000000
14:06:03.630 scripts.js:75 **0x05bbd7e**00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002a30786564453666393442323744344538376135433931623433443346364341343238363336453131324400000000000000000000000000000000000000000000
14:06:03.630 scripts.js:76 **0x05bbd7e**00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002a30786564453666393442323744344538376135433931623433443346364341343238363336453131324400000000000000000000000000000000000000000000


Note the incorrect signatures in bold. Expected ABI to start with **0xffc9896b** & **0x4209fff1** respectively.

**Comment by Muhammad-Altabba** [comment]
Thanks  @TechnicallyWeb3 for reporting this. It should be fixed once https://github.com/web3/web3.js/pull/6922 is merged and it will be available in the version after v4.7.0.
